curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/file-metadata-microservice.english.md

```diff
@@ -23,12 +23,38 @@ Start this project on Repl.it using <a href='https://repl.it/github/freeCodeCamp
 
 ```yml
 tests:
+  - text: I can provide my own project, not the example URL.
+    testString: "getUserInput => {
+      const url = getUserInput('url');
+      assert(!(new RegExp('.*/file-metadata.freecodecamp.repl.co')).test(getUserInput('url')));
+    }"
   - text: I can submit a form that includes a file upload.
-    testString: ''
+    testString: "async getUserInput => {
+      const site = await fetch(getUserInput('url'));
+      const data = await site.text();
+      assert(/<form.*>/.test(data));
+      assert(/type=['|\"]file['|\"]/.test(data));
+    }"
   - text: The form file input field has the <code>name</code> attribute set to <code>upfile</code>.
-    testString: ''
+    testString: "async getUserInput => {
+      const site = await fetch(getUserInput('url'));
+      const data = await site.text();
+      assert(/name=['|\"]upfile['|\"]/.test(data));
+    }"
```

Comment:
What I have here is a bit ugly, but I don't know how comprehensive you want the tests. This is what I have to check that there is an input field and the element with the name 'upfile' has an input field. Some cleaning could be done on mine:

```
const url = getUserInput('url');
        const html =  (await (await fetch(url)).text());
        const doc = new DOMParser().parseFromString(html, 'text/html');
       
        if (doc.getElementsByName('upfile')) {
          assert.isOk(doc.getElementsByName('upfile'), 'There is a tag with name <code>upfile</code>.');
          assert.equal(doc.getElementsByName('upfile').item(0).tagName, 'INPUT');
        } else {
          assert.fail('There are no fields in index.html named <code>upfile</code>')
        }
```